be/src/connector/mysql_connector.h

```diff
@@ -0,0 +1,110 @@
+
+#pragma once
+
+#include "column/column.h"
+#include "column/type_traits.h"
+#include "column/vectorized_fwd.h"
+#include "connector/connector.h"
+#include "exec/mysql_scanner.h"
+
+namespace starrocks {
+
+namespace connector {
+#define APPLY_FOR_NUMERICAL_TYPE(M, APPEND_TO_SQL) \
+    M(TYPE_TINYINT, APPEND_TO_SQL)                 \
+    M(TYPE_BOOLEAN, APPEND_TO_SQL)                 \
+    M(TYPE_SMALLINT, APPEND_TO_SQL)                \
+    M(TYPE_INT, APPEND_TO_SQL)                     \
+    M(TYPE_BIGINT, APPEND_TO_SQL)
+
+#define APPLY_FOR_VARCHAR_DATE_TYPE(M, APPEND_TO_SQL) \
+    M(TYPE_DATE, APPEND_TO_SQL)                       \
+    M(TYPE_DATETIME, APPEND_TO_SQL)                   \
+    M(TYPE_CHAR, APPEND_TO_SQL)                       \
+    M(TYPE_VARCHAR, APPEND_TO_SQL)
+
+class MySQLConnector final : public Connector  {
+public:
+    ~MySQLConnector() override = default;
+
+    DataSourceProviderPtr create_data_source_provider(vectorized::ConnectorScanNode* scan_node,
+                                                      const TPlanNode& plan_node) const override;
+};
+
+class MySQLDataSource;
+class MySQLDataSourceProvider;
+
+class MySQLDataSourceProvider final : public DataSourceProvider {
+public:
+    ~MySQLDataSourceProvider() override = default;
+    friend class MySQLDataSource;
+    MySQLDataSourceProvider(vectorized::ConnectorScanNode* scan_node, const TPlanNode& plan_node);
+    DataSourcePtr create_data_source(const TScanRange& scan_range) override;
+
+    bool insert_local_exchange_operator() const override { return true; }
+    bool accept_empty_scan_ranges() const override { return false; }
+
+protected:
+    vectorized::ConnectorScanNode* _scan_node;
+    const TMySQLScanNode _mysql_scan_node;
+};
+
+class MySQLDataSource final : public DataSource {
+public:
+    ~MySQLDataSource() override = default;
+
+    MySQLDataSource(const MySQLDataSourceProvider* provider, const TScanRange& scan_range);
+    Status open(RuntimeState* state) override;
+    void close(RuntimeState* state) override;
+    Status get_next(RuntimeState* state, vectorized::ChunkPtr* chunk) override;
+
+    int64_t raw_rows_read() const override;
+    int64_t num_rows_read() const override;
+
+private:
+    const MySQLDataSourceProvider* _provider;
+
+    // ============= init func =============
+    Status _init_params(RuntimeState* state);
+
+    // =====================================
+    ObjectPool _obj_pool;
+    ObjectPool* _pool = &_obj_pool;
+    RuntimeState* _runtime_state = nullptr;
+    bool _is_finished = false;
+
+    MysqlScannerParam _my_param;
+    // Name of Mysql table
+    std::string _table_name;
+
+    // Tuple id resolved in prepare() to set _tuple_desc;
+    TupleId _tuple_id;
+
+    // select columns
+    std::vector<std::string> _columns;
+    // where clause
+    std::vector<std::string> _filters;
+    // limit for query with external table.
+    int64_t _limit;
```

Comment:
this variable is stored in `DataSourc` class:
1. conjunct
2. runtime filters
3. read limit